desktop/libs/notebook/src/notebook/connectors/sql_alchemy.py

```diff
@@ -247,22 +251,53 @@ def check_status(self, notebook, snippet):
 
     return response
 
+  @query_error_handler
+  def can_start_over(self, notebook, snippet):
+    guid = snippet['result']['handle']['guid']
```

Comment:
If the user runs the query from the history - it will be the same guid or not? If yes, then the cache will be reused?

Reply:
If the user runs a query from the history, the cache will almost certainly not be there anymore.